content/en/blog/2022-03-21-tools-to-manage-helm-declaratively.md

```diff
@@ -0,0 +1,83 @@
+
+### Other Projects
+
+There are many projects beyond the CNCF projects you can use to help you manage your Helm releases. The following set is an example and not exhaustive.
+
+* [Helmfile](https://github.com/roboll/helmfile) - A declarative spec for deploying Helm charts.
+* [Captain](https://github.com/alauda/captain) - A Helm controller
+* [Helm Provider for Terraform](https://github.com/hashicorp/terraform-provider-helm) - Enables you to manage Helm charts through Terraform
+* [Orkestra](https://azure.github.io/orkestra/) - Release Orchestration and Lifecycle Management (LCM) platform for a related group of Helm releases and their subcharts.
+* [Fleet](https://github.com/rancher/fleet) - A GitOps tool chain that works with Kubernetes manifests, Helm charts, and Kustomize.
+
+### High-Level Tool Comparison
+
+There are some differences between the tools we've looked at so far. The following table provides some insight into their differences. This is not exhaustive and you should evaluate any tools you use yourself.
+
+| | Retains Helm release info | Supports Helm hooks | OCI support |
+| -- | -- | -- | -- |
+| Helmfile | ✅ | ❓[^1] | 🚫 |
+| Terraform Helm provider | ✅ | ✅[^2] | 🚫[^3] |
+| Captain Helm controller | ✅ | ❓[^4] | 🚫[^5] |
+| Flux Helm controller | ✅ | ✅  | 🚫[^6] |
+| ArgoCD | 🚫 | 🚫[^7] | ✅[^8] |
+| Orkestra | ❓ | ❓ | ❓ |
+| Fleet | ✅ | ✅ | 🚫[^9] |
+
+_Note, this comparison is from when the blog post was published. Projects change over time and the feature set may change over time. You should evaluate the projects in their current state before choosing one._
+
+## Conclusion
+
+If you want to use a configuration manager with your Helm and Kubernetes configuration there are many choices. While the Helm project doesn't endorse one project over another, we do suggest using a configuration manager when it's appropriate.
+
+[^1]: Has a custom concept of hooks, not necessarily mapped to Helm hooks. See readme [hooks section](https://github.com/roboll/helmfile#hooks) and [this issue](https://github.com/roboll/helmfile/issues/1291) for clarification and work in progress.
+[^2]: Note there are [some issues](https://github.com/hashicorp/terraform-provider-helm/issues/683) with Helm hooks and wait configurations
+[^3]: See hashicorp/terraform-provider-helm issues [#633](https://github.com/hashicorp/terraform-provider-helm/issues/633), [#666](https://github.com/hashicorp/terraform-provider-helm/issues/666), [#827](https://github.com/hashicorp/terraform-provider-helm/issues/827), [#765](https://github.com/hashicorp/terraform-provider-helm/issues/765), and [#655](https://github.com/hashicorp/terraform-provider-helm/issues/655).
+[^4]: This is not a very widely used project, so unclear which or how well Helm hooks are supported. Scott may want to have another chat with the maintainer (TO-DO: fold in this [reminder code](https://github.com/alauda/captain/blob/master/pkg/helm/printer.go#L23) in a simple way)
```

Comment:
> @scottrigby one of the footnotes says...
> 
> > This is not a very widely used project, so unclear which or how well Helm hooks are supported. Scott may want to have another chat with the maintainer (TO-DO: fold in this [reminder code](https://github.com/alauda/captain/blob/master/pkg/helm/printer.go#L23) in a simple way)
> 
> Did you want to handle that?

@mattfarina ah yes, thanks. I just reached out to Hang Yan to see about their availability to help answer this question. If we hear back and they are available relatively soon, we could wait another few days to publish.

Reply:
@hangyan here are the two questions about Captain are in the PR. In short they are:

1. Which Helm hooks supported by Captain? and how well are they supported?
2. Would you please review the footnote about Captain’s OCI support? If you have any other info about this please let us know.

Your reply would be helpful since you wrote the code. I know you may have to review the code to remind yourself, but our alternative plan is to look at the Captain source code tomorrow, so we can get this post out, but we’d be coming to this code fresh. Thanks in advance 🙂

Reply:
Wrapping up this thread – @hangyan answered by email and @mattfarina updated this PR accordingly 49bd65a576f9d95b0babe497390fc0fa8c9c1bab.